### DISCUSSION TITLE
Apple pay button is not showing.

### DISCUSSION THREAD

**Discussion by Alfaizkhan** [flutter-stripe/flutter_stripe#966] [Q&A] [answered]
**Describe the bug**
Apple pay button is not showing in physical device but working fine in simulator.
Here's code snippet.

      // Initialize the payment sheet
      await Stripe.instance.initPaymentSheet(
        paymentSheetParameters: SetupPaymentSheetParameters(
          // Main params
          paymentIntentClientSecret: clientSecret,
          customerId: customerId,
          customerEphemeralKeySecret: ephemeralKey,
          // Merchant Name
          merchantDisplayName: 'Agent Duty',
          style: ThemeMode.system,
          applePay: const PaymentSheetApplePay(
            merchantCountryCode: 'US',
          ),
        ),
      );

![Simulator Screen Shot - iPhone 11 Pro Max - 2022-09-30 at 21 36 08](https://user-images.githubusercontent.com/33489180/193501096-7dedd7f4-a356-42a0-9ac4-02e6d2cc90f5.png)

**Comment by remonh87** [comment]
did you create a ceritficate as described in :https://developer.apple.com/documentation/passkit/apple_pay/setting_up_apple_pay. ?

**Reply by Alfaizkhan** [reply]
@remonh87 
I did all the configuration in developer account as well as stripe, but still not showing the apply pay button in physical device.

**Comment by Groseuros** [comment]
I had the same issue, fixed once I added a Credit Card to Apple Pay !

**Reply by LiamMckiggan** [reply]
what do you mean a real CC linked to apple pay, you mean a real CC attached to customerID? how could you use a real CC in test mode?

**Reply by orazz** [reply]
> what do you mean a real CC linked to apple pay, you mean a real CC attached to customerID? how could you use a real CC in test mode?

I think you can't test an apple pay in test mode you have to use live mode and real CC.

**Reply by ghost** [reply]
![](https://user-images.githubusercontent.com/4474118/174879244-5b0f9d03-8955-4b48-aac5-c4cb568276c2.png)

**Reply by Groseuros** [reply, marked as answer]
Even in test mode, you need to have Apple Pay enabled and Payment method (with a real / working Card) added to your AppleID on the physical iOS device you are using for test for the Pay button to display.

Settings > Apple ID > Payment & Shipping > Apple Pay > Wallet > Payment Cards

**Reply by remonh87** [reply]
@Groseuros  is correct even in test mode you need to have a real creditcard linked. Luckily stripe won't charge the card and will recognize the paymentintent is paid with a testcode.

**Comment by orazz** [comment]
any solution?

**Comment by ambr89** [comment]
Any solution?

**Reply by Groseuros** [reply]
You need to have a real CC linked to Apple Pay to get it to work on a real device
